### DISCUSSION TITLE
setting role permission for the entire guild

### DISCUSSION THREAD

**Discussion by Mistirio** [Rapptz/discord.py#6111] [Q&A]
How do I take the send messages permission from a role in every channel??

**Comment by Harmon758** [comment]
If you want to set a permission for the entire server, as your title indicates, you can use [`Role.edit`](https://discordpy.readthedocs.io/en/latest/api.html#discord.Role.edit) and pass in a [`Permissions`](https://discordpy.readthedocs.io/en/latest/api.html#discord.Permissions) instance.

For how to get the specific `Role` object, see https://discordpy.readthedocs.io/en/latest/faq.html#how-do-i-get-a-specific-model.

Otherwise, for specific text channels, you can use [`TextChannel.set_permissions`](https://discordpy.readthedocs.io/en/latest/api.html#discord.TextChannel.set_permissions) or  [`TextChannel.edit`](https://discordpy.readthedocs.io/en/latest/api.html#discord.TextChannel.edit) and [`PermissionOverwrite`](https://discordpy.readthedocs.io/en/latest/api.html#discord.PermissionOverwrite).
In this case, the pertinent permission would be [`Permissions.send_message`](https://discordpy.readthedocs.io/en/latest/api.html#discord.Permissions.send_messages).

**Reply by Mistirio** [reply]
Ok thanks
